packages/go_router/test/configuration_test.dart

```diff
@@ -0,0 +1,250 @@
+            ],
+            redirectLimit: 10,
+            topRedirect: (GoRouterState state) {
+              return null;
+            },
+          );
+        },
+        throwsAssertionError,
+      );
+    });
+
+    test(
+        'throws when there is a GoRoute ancestor with a different parentNavigatorKey',
+        () {
+      final GlobalKey<NavigatorState> root =
+          GlobalKey<NavigatorState>(debugLabel: 'root');
+      final GlobalKey<NavigatorState> shell =
+          GlobalKey<NavigatorState>(debugLabel: 'shell');
+      expect(
+        () {
+          RouteConfiguration(
+            navigatorKey: root,
+            routes: <RouteBase>[
+              ShellRoute(
+                navigatorKey: shell,
+                routes: <RouteBase>[
+                  GoRoute(
+                    path: '/',
+                    builder: _mockScreenBuilder,
+                    parentNavigatorKey: root,
+                    routes: <RouteBase>[
+                      GoRoute(
+                        path: 'a',
+                        builder: _mockScreenBuilder,
+                        parentNavigatorKey: shell,
+                      ),
+                    ],
+                  ),
+                ],
+              ),
+            ],
+            redirectLimit: 10,
+            topRedirect: (GoRouterState state) {
+              return null;
+            },
+          );
+        },
+        throwsAssertionError,
+      );
+    });
+
+    test(
+      'Does not throw with valid parentNavigatorKey configuration',
+      () {
+        final GlobalKey<NavigatorState> root =
+            GlobalKey<NavigatorState>(debugLabel: 'root');
+        final GlobalKey<NavigatorState> shell =
+            GlobalKey<NavigatorState>(debugLabel: 'shell');
+        final GlobalKey<NavigatorState> shell2 =
+            GlobalKey<NavigatorState>(debugLabel: 'shell2');
+        RouteConfiguration(
+          navigatorKey: root,
+          routes: <RouteBase>[
+            ShellRoute(
+              navigatorKey: shell,
+              routes: <RouteBase>[
+                GoRoute(
+                  path: '/',
+                  builder: _mockScreenBuilder,
+                  routes: <RouteBase>[
+                    GoRoute(
+                      path: 'a',
+                      parentNavigatorKey: root,
+                      builder: _mockScreenBuilder,
+                      routes: <RouteBase>[
+                        ShellRoute(
+                          navigatorKey: shell2,
+                          routes: <RouteBase>[
+                            GoRoute(
+                              path: 'b',
+                              builder: _mockScreenBuilder,
+                              routes: <RouteBase>[
+                                GoRoute(
+                                  path: 'b',
+                                  builder: _mockScreenBuilder,
+                                  parentNavigatorKey: shell,
```

Comment:
shouldn't this fail? the `ShellRoute(navigatorKey: shell)` will be beneath `GoRoute(path: 'a', parentNavigatorKey: root)`

that means the GoRoute(path: 'b') will not be placed in the top-most page

Reply:
Good catch, we were allowing GoRouteAncestors to have a different parentNavigatorKey if there is a ShellRoute in between, but that could be wrong. Fixed: 2f4bc7dee6b6a5e8506182d3cf42b999de7c2c27